lib/cors-anywhere.js

```diff
@@ -364,6 +365,12 @@ function getHandler(options, proxy) {
       return;
     }
 
+    if (corsAnywhere.checkTargetWhitelisted && !corsAnywhere.checkTargetWhitelisted(location)) {
```

Comment:
This is not the right place to enforce target restrictions, because it does not account for redirects.

I once started with a patch to limit targets, in #78, but never finished it.